iothub/service/src/Registry/DevicesClient.cs

```diff
@@ -1147,7 +950,7 @@ private static IEnumerable<ExportImportDevice> GenerateExportImportDeviceListFor
                     case ImportMode.UpdateIfMatchETag:
                         if (string.IsNullOrWhiteSpace(device.ETag.ToString()))
```

Comment:
@timtay-microsoft @abhipsaMisra what do you think about removing this (and delete if match etag) validation is removed (like we have for non-bulk operations)?

To me, the downside here is that they won't get immediate errors, rather they have to check the errors blob, which is much more costly.

Reply:
I'm fine with keeping this validation here for the reasons you outlined above. In the non-bulk APIs, it is easy enough for the customer to handle a 412

Reply:
+1 to that. No point in increasing an additional out-of-the-way step for the user to discover the error.